metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -165,7 +164,8 @@ private ConfigurationControlManager(LogContext logContext,
             ConfigurationValidator validator,
             Map<String, Object> staticConfig,
             int nodeId,
-            FeatureControlManager featureControl) {
```

Comment:
Can we just have a reference to `FeatureControlManager` inside `ConfigurationControlManager`? We have that in other places and I think it makes sense.

Feature control is the "bottom turtle" for all this stuff since things behave differently based on features etc.